virtcontainers/kata_agent.go

```diff
@@ -1706,6 +1718,23 @@ func (k *kataAgent) handlePidNamespace(grpcSpec *grpc.Spec, sandbox *Sandbox) bo
 	return sharedPidNs
 }
 
+// handlePidNamespace checks if Pid namespace for a container needs to be shared with its sandbox
+// pid namespace. This function also modifies the grpc spec to remove the pid namespace
+// from the list of namespaces passed to the agent.
+func (k *kataAgent) checkAgentPidNs(container *Container) bool {
+	agentPidNs := false
+
+	for _, env := range container.config.Cmd.Envs {
+		if env.Var == agentPidEnv {
```

Comment:
ehmm this is prone to vulnerabilities, isn't it? anyone will be able to join the agent's pid namespace,

Reply:
A container can join agent pid namespace only if this feature is explicitly enabled in the configuration.toml, which only a root user will be able to do.